server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -173,6 +268,38 @@ export function verifyStorageToken(
 				new NetworkError(403, "Missing documentId in request"),
 			);
 		}
+
+		// TODO: remove this check and code after this block after validation
+		if (options.enableTokenCache) {
+			const moreOptions: IVerifyTokenOptions = options;
+			moreOptions.maxTokenLifetimeSec = maxTokenLifetimeSec;
+			moreOptions.requireTokenExpiryCheck = isTokenExpiryEnabled;
+			try {
+				await verifyToken(
```

Comment:
This bypasses single-use token verification, which creates a security flaw in our create document flow.

Reply:
This verifyToken include single token check: line 195

Reply:
My idea here is to use verifyToken in the verifyStorageToken function and remove the rest logic. Then this verifyToken can be reused in historian since it is not a middleware anymore and historian need to get the token in a different way

Reply:
Got it, I didn't realize you had moved it, and just saw that the old codepath was bypassed when enableTokenCache was true. Thanks for pointing me in the right direction. Historian doesn't need the `ensureSingleUseToken` functionality, but it should be fine to share and it won't hurt anything. Thanks!